engine/minskyTensorOps.cc

```diff
@@ -433,6 +433,38 @@ namespace minsky
         hc.xvectors.insert(hc.xvectors.begin(), xv2.begin()+1, xv2.end());        
         hc.xvectors.insert(hc.xvectors.begin(), xv1.begin(), xv1.end()-1);
         cachedResult.hypercube(move(hc));
+        
+        set<size_t> newIdx;
+        vector<size_t> hcIdx1,hcIdx2;       
+        
+        if (!arg1->index().empty() && !arg2->index().empty())
+        {
+        for (auto& i: arg1->index()) 
+			for (auto& k: arg1->hypercube().splitIndex(i))
+                 hcIdx1.push_back(k);          
+	
+        for (auto& j: arg2->index())   
+            for (auto& l: arg2->hypercube().splitIndex(j)) 
+                 hcIdx2.push_back(l);
+        
+	    vector<size_t> tmpIdx;
+	    size_t tmpLinealIdx;        
+        
+        // first element of split index of first tensor and last element of split index of second tensor is the split index of product tensor. Convert to lineal index to get new index vector????    
+        for (size_t i=0; i< min(hcIdx1.size(),hcIdx2.size());i++)
+          {
+			 if (i==0) tmpIdx.push_back(hcIdx1[i]);
+			 if (i == min(hcIdx1.size(),hcIdx2.size())-1) tmpIdx.push_back(hcIdx2[i]);
+		 }
```

Comment:
I don't think this is correct. Yes - the approach is valid, but the new split index should be something like [hcIdx1[0],...hcIdx1[hcIdx.size()-2,hcIdx2[1],...hcIdx2[hcIdx2.size()-1]]. You can use insert operations:

vector<size_t> tmpIdx(hcIdx1.begin(), hcIdx1.end()-1);
tmpIdx.insert(tmpIdx.end(),hcIdx2.begin()+1,hcIdx2.end());